drivers/gpu/drm/vc4/vc4_plane.c

```diff
@@ -458,14 +458,46 @@ static void vc4_write_tpz(struct vc4_plane_state *vc4_state, u32 src, u32 dst)
 			VC4_SET_FIELD(recip, SCALER_TPZ1_RECIP));
 }
 
-static void vc4_write_ppf(struct vc4_plane_state *vc4_state, u32 src, u32 dst)
+#define PHASE_BITS 6
```

Comment:
Phase magnitude bits. There's a sign bit on top.

I was having to think what lines 493-496 were doing, as they seemed to be duplication. Ensure +ve values don't touch the sign bit, then truncate negative values.

It'd be worth a comment to that effect.